lang/golang2mgmt/config.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"fmt"
+	"log"
+)
+
+type config struct {
+	Functions functions `yaml:"functions"`
+}
+
+type functions []function
+
+type testarg struct {
+	Name  string `yaml:"name",omitempty`
+	Type  string `yaml:"type"`
+	Value string `yaml:"value"`
+}
+
+type arg struct {
+	Name string `yaml:"name",omitempty`
+	Type string `yaml:"type"`
+}
+
+// ToMcl prints the arg signature as expected by mcl.
+func (obj *arg) ToMcl() string {
+	if obj.Type == "string" {
+		if obj.Name != "" {
+			return fmt.Sprintf("%s str", obj.Name)
+		}
+		return "str"
+	}
+	log.Fatalf("Can not convert %v to mcl", obj)
```

Comment:
(Here and elsewhere)

I have to be annoying and enforce cleaner code, even if it's in this small util program-- Please have proper function signatures like `(string, error)`, and propagate the errors upwards... Only at the top-level do we run log.Whatever()

Also errors start with a lower case.

You can use `errwrap.Wrapf` to wrap errors upwards if needed.

Lastly it's "cannot" :)